Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/ThresholdedRandomCutForestState.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.parkservices.threshold;
+
+import static com.amazon.randomcutforest.state.Version.V2_1;
+
+import lombok.Data;
+
+import com.amazon.randomcutforest.parkservices.threshold.state.BasicThresholderState;
+import com.amazon.randomcutforest.returntypes.DiVector;
+import com.amazon.randomcutforest.state.RandomCutForestState;
+
+@Data
+public class ThresholdedRandomCutForestState {
+    private String version = V2_1;
+    RandomCutForestState forestState;
+    private BasicThresholderState thresholderState;
+    private double ignoreSimilarFactor;
+    private double triggerFactor;
+    private long lastAnomalyTimeStamp;
+    private double lastAnomalyScore;
+    private DiVector lastAnomalyAttribution;
```

Comment:
issue. The serialization library cannot handle this class due to its constructor and data. Either this state uses a more primitive structure or the divector class is changed to be more like a plain POJO.

Reply:
Maybe a better engineering can focus on making these data structures POJO? RCF library is not in the business of serialization. That being said, a Divector is a collection of primitive arrays. We should refrain from using "issues" for simpler concerns.

Reply:
Fixed.